pkg/internal/ebpf/common/kafka_detect_transform.go

```diff
@@ -206,23 +206,25 @@ func getTopicName(pkt []byte, offset int, op Operation, apiVersion int16) (strin
 		maxLen = len(pkt)
 	}
 	topicName := pkt[offset:maxLen]
-	if op == Fetch && apiVersion > 11 { // topic name has the following format: uuid\x00\x02\tTOPIC\x02\x00
-		topicNames := strings.Split(string(topicName), "\t")
-		if len(topicNames) >= 2 {
-			topicNames := strings.Split(string(topicNames[1]), "\x02\x00")
-			if len(topicNames) > 0 {
-				topicName = []byte(topicNames[0])
-			}
-		} else {
-			topicName = []byte(topicNames[0])
-		}
+	if op == Fetch && apiVersion > 11 {
+		// topic name has the following format: uuid\x00\x02\tTOPIC\x02\x00
+		topicName = []byte(extractTopic(string(topicName)))
 	}
 	if isValidKafkaString(topicName, len(topicName), int(topicNameSize), false) {
 		return string(topicName), nil
 	}
 	return "", errors.New("invalid topic name")
 }
 
+func extractTopic(input string) string {
+	re := regexp.MustCompile("\x02\t(.*)\x02")
```

Comment:
compiling a regexp on each usage might have a noticeable impact on performance.  I'd move the regexp compilation to a global variable:
```go
var re = regexp.MustCompile("\x02\t(.*)\x02")
func extractTopic(input string) string {
```

Reply:
ah yeah, right